folium/features.py

```diff
@@ -682,7 +687,156 @@ def get_bounds(self):
         ]
 
 
-class GeoJsonTooltip(Tooltip):
+class GeoJsonDetail(MacroElement):
+
+    """
+    Base class for GeoJsonTooltip and GeoJsonPopup to inherit methods and
+    template structure from. Not for direct usage.
+
+    """
+    base_template = u"""
+    function(layer){
+    let div = L.DomUtil.create('div');
+    {% if this.fields %}
+    let handleObject = feature=>typeof(feature)=='object' ? JSON.stringify(feature) : feature;
+    let fields = {{ this.fields | tojson | safe }};
+    let aliases = {{ this.aliases | tojson | safe }};
+    let table = '<table>' +
+        String(
+        fields.map(
+        (v,i)=>
+        `<tr>{% if this.labels %}
+            <th>${aliases[i]{% if this.localize %}.toLocaleString(){% endif %}}</th>
+            {% endif %}
+            <td>${handleObject(layer.feature.properties[v]){% if this.localize %}.toLocaleString(){% endif %}}</td>
+        </tr>`).join(''))
+    +'</table>';
+    div.innerHTML=table;
+    {% endif %}
+    {% if this.vegalite %}
+    let mapValue = layer.feature.properties['{{ this.map_key }}']
+    let dataKey = '{{this.data_key }}'
+    Object.keys({{ this._parent.get_name() }}datasets).forEach(function(k){
+    {{ this._parent.get_name() }}spec['datasets'][k]={{ this._parent.get_name() }}datasets[k].filter(
+        o=>o[dataKey]===mapValue)
+    })
+    let chartembed = L.DomUtil.create('div','foliumchart')
+    div.appendChild(chartembed)
+    vegaEmbed(chartembed, {{ this._parent.get_name()}}spec)
+    {% endif %}
+    return div
+    }
+    """
+
+    def __init__(self, fields, aliases=None, labels=True, localize=False, style=None, class_name="geojsondetail",
+                 vegalite=None, map_key=None, data_key=None):
+        super(GeoJsonDetail, self).__init__()
+        assert isinstance(fields, (list, tuple)), 'Please pass a list or ' \
+                                                  'tuple to fields.'
+        if aliases is not None:
+            assert isinstance(aliases, (list, tuple))
+            assert len(fields) == len(aliases), 'fields and aliases must have' \
+                                                ' the same length.'
+        assert isinstance(labels, bool), 'labels requires a boolean value.'
+        assert isinstance(localize, bool), 'localize must be bool.'
+        self._name = "GeoJsonDetail"
+        self.fields = fields
+        self.aliases = aliases if aliases is not None else fields
+        self.labels = labels
+        self.localize = localize
+        self.class_name = class_name
```

Comment:
@Conengmo - sorry for the extended delay on the modifications to this PR - I think I've addressed most of your concerns with this revision.

re: the `class_name`: I thought it might be useful for users wanting to add custom css declarations to style the tooltips with a human-readable name. It's also allowed us to take the inline styling out of the html elements and add a top level styling to the class in the `head` of the document.